torch/distributions/multivariate_normal.py

```diff
@@ -184,9 +184,11 @@ def covariance_matrix(self):
 
     @lazy_property
     def precision_matrix(self):
-        identity = torch.eye(self.loc.size(-1), device=self.loc.device, dtype=self.loc.dtype)
+        # identity = torch.eye(self.loc.size(-1), device=self.loc.device, dtype=self.loc.dtype)
         # TODO: use cholesky_inverse when its batching is supported
-        return torch.cholesky_solve(identity, self._unbroadcasted_scale_tril).expand(
+        # return torch.cholesky_solve(identity, self._unbroadcasted_scale_tril).expand(
+        #     self._batch_shape + self._event_shape + self._event_shape)
+        return torch.cholesky_inverse(self._unbroadcasted_scale_tril).expand(
```

Comment:
I'm not sure if `torch.cholesky_inverse` supports batched tensors. Have you verified that this works for tensors of shape `(b, n, n)`?

Reply:
Yes, I checked out empirically, which is not currently marked in docs.
```
>>> print(torch.__version__)
1.10.0+cu113
>>> a = torch.randn(2, 3, 3)
>>> a = torch.einsum("abc,adc->abd", a, a)
>>> u = torch.linalg.cholesky(a)
>>> u
tensor([[[ 0.5249,  0.0000,  0.0000],
         [-1.2942,  0.7440,  0.0000],
         [-0.2534, -1.0307,  0.9120]],

        [[ 1.5023,  0.0000,  0.0000],
         [ 1.0731,  1.1105,  0.0000],
         [ 0.1236, -0.1171,  1.3490]]])
>>> torch.cholesky_inverse(u)
tensor([[[32.8829, 10.9476,  4.6869],
         [10.9476,  4.1142,  1.6656],
         [ 4.6869,  1.6656,  1.2022]],

        [[ 0.8705, -0.5884, -0.0866],
         [-0.5884,  0.8170,  0.0579],
         [-0.0866,  0.0579,  0.5495]]])
>>> a.inverse()
tensor([[[32.8829, 10.9476,  4.6869],
         [10.9476,  4.1142,  1.6656],
         [ 4.6869,  1.6656,  1.2022]],

        [[ 0.8705, -0.5884, -0.0866],
         [-0.5884,  0.8170,  0.0579],
         [-0.0866,  0.0579,  0.5495]]])
```

Reply:
Good observation! I see that it was implemented in #50269 but the docs weren't updated to reflect this.

Reply:
If no one is trying to update it, I will reflect it to the doc.

Reply:
Thanks for volunteering to fix the docs. If it is useful, you could refer to this PR (https://github.com/pytorch/pytorch/pull/31275) by @vishwakftw. This wasn't merged but he had changed the docstring to reflect batched inputs in `cholesky_inverse`. Let us do this in another PR though since it is unrelated to your changes here.

Reply:
@neerajprad Thanks for your suggestion. I am trying to make another PR and I wonder how I can request you as the reviewer. Shoule I just write PR # here after making new?

Reply:
Just cc me on the PR and I'll take a look.